src/cmd/services/m3comparator/main/querier.go

```diff
@@ -147,15 +155,39 @@ func (q *querier) FetchCompressed(
 	}, cleanup, nil
 }
 
-func (q *querier) generateRandomIters(
+func (q *querier) generateRandomSeries(
 	query *storage.FetchQuery,
-) (encoding.SeriesIterators, error) {
+	blockSize time.Duration,
+) ([]series, error) {
+	var (
+		start = query.Start.Truncate(blockSize)
+		end   = query.End.Truncate(blockSize).Add(blockSize)
+	)
+
+	multiSeriesMetrics := ""
+	for _, matcher := range query.TagMatchers {
+		if bytes.Equal(q.opts.tagOptions.MetricName(), matcher.Name) {
```

Comment:
nit: assign `q.opts.tagOptions.MetricName()` outside of this loop since it won't change? Alternatively just extract and store the `MetricName` in `querier`

Reply:
Also, do you want to key this on the `__name__`? May be worth adding a new `multi` tag to key this on, so that it can maintain a given name? Up to you, whatever's easier to work with on the other side is fine

Reply:
An interesting point, need to think about this.